AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSAnalyticsAuthenticationProvider.m

```diff
@@ -0,0 +1,84 @@
+
+#import "MSAnalyticsAuthenticationProvider.h"
+
+#import "MSAnalyticsInternal.h"
+#import "MSLogger.h"
+#import "MSTicketCache.h"
+#import "MSUtility+StringFormatting.h"
+#import "MSAnalyticsAuthenticationResult.h"
+
+static int kMSRefreshThreshold = 10 * 60;
+
+@interface MSAnalyticsAuthenticationProvider ()
+
+@property(nonatomic) BOOL isAlreadyAcquiringToken;
+
+@property(nonatomic) NSDate *expiryDate;
+
+@end
+
+@implementation MSAnalyticsAuthenticationProvider
+
+- (instancetype)initWithAuthenticationType:(MSAnalyticsAuthenticationType)type
+                                 ticketKey:(NSString *)ticketKey
+                         completionHandler:
+                             (MSAcquireTokenCompletionBlock)completionHandler {
+  if ((self = [super init])) {
+    _type = type;
+    _ticketKey = ticketKey;
+    if (ticketKey) {
+      _ticketKeyHash = [MSUtility sha256:ticketKey];
+    }
+    _completionHandler = completionHandler;
+  }
+  return self;
+}
+
+- (void)acquireTokenAsync {
+  if (self.completionHandler) {
```

Comment:
Do we need to add some synchronization here?

Reply:
I had the same question during pairing and we could not find an issue with this code. Can you?